ui-core/src/main/java/com/adyen/checkout/ui/core/internal/ui/DefaultAddressLookupDelegate.kt

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 22/12/2023.
+ */
+
+package com.adyen.checkout.ui.core.internal.ui
+
+import androidx.annotation.RestrictTo
+import androidx.annotation.VisibleForTesting
+import com.adyen.checkout.components.core.AddressInputModel
+import com.adyen.checkout.components.core.AddressLookupCallback
+import com.adyen.checkout.components.core.AddressLookupResult
+import com.adyen.checkout.components.core.LookupAddress
+import com.adyen.checkout.components.core.internal.util.bufferedChannel
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.internal.util.LogUtil
+import com.adyen.checkout.core.internal.util.Logger
+import com.adyen.checkout.ui.core.internal.data.api.AddressRepository
+import com.adyen.checkout.ui.core.internal.ui.model.AddressListItem
+import com.adyen.checkout.ui.core.internal.ui.model.AddressLookupEvent
+import com.adyen.checkout.ui.core.internal.ui.model.AddressLookupInputData
+import com.adyen.checkout.ui.core.internal.ui.model.AddressLookupState
+import com.adyen.checkout.ui.core.internal.ui.model.AddressOutputData
+import com.adyen.checkout.ui.core.internal.ui.model.AddressParams
+import com.adyen.checkout.ui.core.internal.ui.view.LookupOption
+import com.adyen.checkout.ui.core.internal.util.AddressFormUtils
+import com.adyen.checkout.ui.core.internal.util.AddressValidationUtils
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.receiveAsFlow
+import java.util.Locale
+
+@Suppress("TooManyFunctions")
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class DefaultAddressLookupDelegate(
+    private val addressRepository: AddressRepository,
+    private val shopperLocale: Locale
+) :
+    AddressLookupDelegate,
+    AddressDelegate {
+
+    private var coroutineScope: CoroutineScope? = null
+
+    override val addressDelegate: AddressDelegate = this
+
+    private var addressLookupCallback: AddressLookupCallback? = null
+
+    private val addressLookupInputData = AddressLookupInputData()
+
+    @VisibleForTesting
+    internal val mutableAddressLookupStateFlow = MutableStateFlow<AddressLookupState>(AddressLookupState.Initial)
```

Comment:
Nitpick: we'd use `_addressLookupStateFlow` in other places

Reply:
i can't do that for non private values